src/devices/Mcp25xxx/Register/MessageReceive/RxBxSidl.cs

```diff
@@ -0,0 +1,158 @@
+        /// </param>
+        /// <param name="ide">
+        /// Extended Identifier Flag bit.
+        /// This bit indicates whether the received message was a standard or an extended frame.
+        /// True = Received message was an extended frame.
+        /// False = Received message was a standard frame.
+        /// </param>
+        /// <param name="srr">
+        /// Standard Frame Remote Transmit Request bit (valid only if the IDE bit = 0).
+        /// True = Standard frame Remote Transmit Request received.
+        /// False = Standard data frame received.
+        /// </param>
+        /// <param name="sid">
+        /// Extended Identifier bits.
+        /// </param>
+        public RxBxSidl(RxBufferNumber rxBufferNumber, byte eid, bool ide, bool srr, byte sid)
+        {
+            if (eid > 3)
+            {
+                throw new ArgumentException($"Invalid EID value {eid}.", nameof(eid));
+            }
+
+            if (sid > 7)
+            {
+                throw new ArgumentException($"Invalid SID value {sid}.", nameof(sid));
+            }
+
+            RxBufferNumber = rxBufferNumber;
+            Eid = eid;
+            Ide = ide;
+            Srr = srr;
+            Sid = sid;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the RxBxSidl class.
+        /// </summary>
+        /// <param name="value">The value that represents the register contents.</param>
+        public RxBxSidl(RxBufferNumber rxBufferNumber, byte value)
+        {
+            RxBufferNumber = rxBufferNumber;
+            Eid = (byte)(value & 0b0000_0011);
+            Ide = (value & 0b0000_1000) == 0b0000_1000;
+            Srr = (value & 0b0001_0000) == 0b0001_0000;
+            Sid = (byte)((value & 0b1110_0000) >> 5);
+        }
+
+        /// <summary>
+        /// Receive Buffer Number.
+        /// </summary>
+        public RxBufferNumber RxBufferNumber { get; }
+
+        /// <summary>
+        /// Extended Identifier bits.
+        /// These bits contain the three Least Significant bits of the Standard Identifier for the received message.
+        /// </summary>
+        public byte Eid { get; }
+
+        /// <summary>
+        /// Extended Identifier Flag bit.
+        /// This bit indicates whether the received message was a standard or an extended frame.
+        /// True = Received message was an extended frame.
+        /// False = Received message was a standard frame.
+        /// </summary>
+        public bool Ide { get; }
+
+        /// <summary>
+        /// Standard Frame Remote Transmit Request bit (valid only if the IDE bit = 0).
+        /// True = Standard frame Remote Transmit Request received.
+        /// False = Standard data frame received.
+        /// </summary>
+        public bool Srr { get; }
+
+        /// <summary>
+        /// Extended Identifier bits.
+        /// </summary>
+        public byte Sid { get; }
```

Comment:
I'm wondering - Extended identifier should be 29-bit but this seems to be at most 16 - I'm slightly confused here - I'd double check if this is correct (you previously mentioned you had some issues with getting this to work - perhaps this is a culprit)

Reply:
@krwq Standard is 11-bit.  This is for the first 3 lower bits SID[2:0].  And RxBxSidh is for the top 8 [10:3].  I see my comments might be off (another copy/paster :smile: ).   I noticed when working on the summaries the datasheet wasn't consistent for these types of parameters, where some pointed the bits out and others didn't.  I'll try to review and update those to make the same, as well.

Reply:
@shaggygi I know - comment says extended but name standard :smile:

Reply:
Fixed